bigquery_etl/cli/query.py

```diff
@@ -1349,94 +1362,118 @@ def initialize(ctx, name, sql_dir, project_id, dry_run, parallelism):
         )
         sys.exit(1)
 
-    for query_file in query_files:
-        sql_content = query_file.read_text()
+    def _initialize(query_file):
+        project, dataset, destination_table = extract_from_query_path(query_file)
+        client = bigquery.Client(project=project)
+        full_table_id = f"{project}.{dataset}.{destination_table}"
+        table = None
 
-        # Enable initialization from query.sql files
-        # Create the table by deploying the schema and metadata, then run the init.
-        # This does not currently verify the accuracy of the schema or that it
-        # matches the query.
-        if "is_init()" in sql_content:
-            project = query_file.parent.parent.parent.name
-            dataset = query_file.parent.parent.name
-            destination_table = query_file.parent.name
-            full_table_id = f"{project}.{dataset}.{destination_table}"
+        sql_content = query_file.read_text()
+        init_files = list(Path(query_file.parent).rglob("init.sql"))
 
+        # check if the provided file can be initialized and whether existing ones should be skipped
+        if "is_init()" in sql_content or len(init_files) > 0:
             try:
                 table = client.get_table(full_table_id)
-                if table.num_rows > 0:
+                if skip_existing:
+                    # table exists; skip initialization
+                    return
+                if not force and table.num_rows > 0:
```

Comment:
Added a `--force` option. I think it's also fine to use this to overwrite materialized views.